### PR TITLE
enable PathTooLongException test

### PR DESCRIPTION
**Pull request by zkat**
Fixes: https://github.com/NuGet/Home/issues/8920

Frankly, idk why that comment was there. This test seems to pass just fine without accounting for the `*` thing. Let's see what CI says, though!

### CODE REVIEW

File: test/NuGet.Core.FuncTests/NuGet.Commands.FuncTest/RestoreCommandTests.cs [hunk 1]
```diff
@@ -1855,8 +1855,6 @@ public async Task RestoreCommand_LockedLockFileWithOutOfDateProjectAsync()
             }
         }
 
-#if IS_DESKTOP
-        // TODO: To work on coreclr we need to address https://github.com/NuGet/Home/issues/7588
```
Comment (nkolev92) [inline test/NuGet.Core.FuncTests/NuGet.Commands.FuncTest/RestoreCommandTests.cs:1859]: I think .NET Core solved this `problem`. 
If the OS supports it ofc. 

The max path length is now shortmax but there might be restrictions about certain segments. 

tldr; 

I don't think you can get it to throw on .NET Core if the machine has long path support enabled (automatic on latest windows versions)

Comment (zkat) [reply test/NuGet.Core.FuncTests/NuGet.Commands.FuncTest/RestoreCommandTests.cs:1859]: I've pushed a new change that skips testing this if long paths are enabled on the current machine. Does that seem like the right way to go about this?

Comment (nkolev92) [reply test/NuGet.Core.FuncTests/NuGet.Commands.FuncTest/RestoreCommandTests.cs:1859]: Yeah, I think that's ok. 

Should this change be targeting dev instead of the feature branch?

**Review by heng-liu — APPROVED** [review]

**Review by nkolev92 — APPROVED** [review]
